pulp_docker/app/token_verification.py

```diff
@@ -0,0 +1,155 @@
+import jwt
+
+from aiohttp import web
+from django.conf import settings
+
+
+class TokenVerifier:
+    """A class used for a token verification."""
+
+    def __init__(self, request, access_action):
+        """
+        Store data required for the token verification.
+
+        Args:
+            request (:class:`~aiohttp.web.Request`): The request with an Authorization header.
+            access_action (str): A required action to perform pulling/pushing.
+
+        """
+        self.request = request
+        self.access_action = access_action
+
+    def verify(self):
+        """Verify a Bearer token."""
+        authorization = self.get_authorization_header()
+        self.check_authorization_token(authorization)
+
+    def get_authorization_header(self):
```

Comment:
auth header should contain proper content-type header as well as Docker-Distribution-Api-Version
see:
```
http https://registry-1.docker.io/v2/
HTTP/1.1 401 Unauthorized
Content-Length: 87
Content-Type: application/json
Date: Tue, 01 Oct 2019 11:19:11 GMT
Docker-Distribution-Api-Version: registry/2.0
Strict-Transport-Security: max-age=31536000
Www-Authenticate: Bearer realm="https://auth.docker.io/token",service="registry.docker.io"
```